server/controllers/match/search/bedSearchController.ts

```diff
@@ -43,7 +43,7 @@ export default class BedSearchController {
       const placementRequest = await this.placementRequestService.getPlacementRequest(req.user.token, req.params.id)
       const searchParams = mapPlacementRequestToBedSearchParams(placementRequest)
 
-      const query = objectIfNotEmpty<BedSearchParametersUi>(searchParams)
+      const query = objectIfNotEmpty<BedSearchParametersUi>(searchParams as unknown as Record<string, unknown>)
```

Comment:
I think we could improve things here by changing the param type of `object` that `objectIfNotEmpty` accepts to be `object: Record<string, unknown> | T | undefined`. 
This way we can remove the `[index: string]: unknown` from `BedSearchParametersUi` *and* avoid casting here. My bad I should've realised when I wrote the function originally.

Reply:
Nice - good catch 👍